hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -37,31 +37,132 @@
     /*
         CRUD interface for Trigger
      */
+
+    /**
+     * Create a new <code>Trigger</code>.  <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The new </code>Trigger</code> will be persisted.  When fully defined a call to
+     * {@link #updateTrigger(Trigger)}
+     * is needed to enable the </code>Trigger</code>.
+     * @param trigger
+     * @throws Exception If the </code>Trigger</code> already exists.
+     */
     void addTrigger(Trigger trigger) throws Exception;
+
+    /**
+     * The <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be persisted.
+     * @param triggerId
+     * @throws Exception
+     */
     void removeTrigger(String triggerId) throws Exception;
-    void updateTrigger(Trigger trigger) throws Exception;
-    Collection<Trigger> getTriggers() throws Exception;
+
+    /**
+     * Update the <code>Trigger</code>. <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The updated </code>Trigger</code> will be persisted.  If enabled the </code>Trigger</code>
+     * will be [re-]inserted into the Alerts engine and any prior dampening will be reset.
+     * @param triggerId
+     * @throws Exception If the </code>Trigger</code> does not exist.
+     */
+    Trigger updateTrigger(Trigger trigger) throws Exception;
+
     Trigger getTrigger(String triggerId) throws Exception;
 
+    Collection<Trigger> getAllTriggers() throws Exception;
+
     /*
-        CRUD interface for Trigger
+        CRUD interface for Dampening
+     */
+
+    Dampening addDampening(Dampening dampening) throws Exception;
+
+    void removeDampening(String dampeningId) throws Exception;
+
+    Dampening updateDampening(Dampening dampening) throws Exception;
+
+    Dampening getDampening(String dampeningId) throws Exception;
```

Comment:
With the Dampening class I have also a main doubt: does it need to be part of the Trigger definition itself ?
I mean, from a public API perspective, all interesting info is in 1 to 1 relation with the Trigger class.
Dampening class also has transient info that is only valid for the internal inference inside the engine, but it should not visible from an API perspective IMO.
So, from a pure API perspective I'm wondefing if we should move public data of Dampening object into the Trigger and keep Dampening object but not in public api model, but in a private model that is interesting only for the engine.